staging/src/k8s.io/cri-api/README.md

```diff
@@ -6,9 +6,194 @@ without the need to recompile. CRI consists of a protocol buffers and gRPC API.
+the version skew policy.
+
+### Version Skew Policy for CRI API
+
+CRI API has two versions:
+- Major semantic version (known versions are `v1alpha2` ([removed in 1.26](https://kubernetes.io/blog/2022/12/09/kubernetes-v1-26-release/#cri-v1alpha2-removed)), `v1`).
+- Kubernetes version (for example: `@1.23`). Note, the `cri-api` Golang library is versioned as `0.23` as it doesn't guarantee Go types backward compatibility.
+
+Major semantic version (e.g. `v1`) is used to introduce breaking changes
+and major new features that are incompatible with the current API.
+
+Kubernetes version is used to indicate a specific feature set implemented
+on top of the major semantic version. All changes made without the change
+of a major semantic version API must be backward and forward compatible.
+
+- _Kubelet_ must work with the older _Container Runtime_ if it implements
+  the same semantic version of CRI API (e.g. `v1`) of up to three Kubernetes minor
+  versions back. New features implemented in CRI API must be gracefully degraded.
+  For example, _Kubelet_ of version 1.26 must work with _Container Runtime_
+  implementing `k8s.io/cri-api@v0.23.0`+.
+- _Kubelet_ must work with _Container Runtime_ if it implements
+  the same semantic version of CRI API (e.g. `v1`) of up to
+  three minor versions up. New features implemented in CRI API must not change
+  behavior of old method calls and response values. For example, _Kubelet_ of
+  version 1.22 must work with _Container Runtime_ implementing `k8s.io/cri-api@v0.25.5`.
+
+
+## Versioning
+
+This library contains go classes generated from the CRI API protocol buffers and gRPC API.
+
+The library versioned as `0.XX` as Kubernetes doesn't provide any guarantees
+on backward compatibility of Go wrappers between versions. However CRI API itself
+(protocol buffers and gRPC API) is marked as stable `v1` version and it is
+backward compatible between versions.
+
+Versions like `v0.<minor>.<patch>` (e.g. `v0.25.5`) are considered stable.
+It is discouraged to introduce CRI API changes in patch releases and recommended
+to use versions like `v0.<minor>.0`.
+
+All alpha and beta versions (e.g. `k8s.io/cri-api@v0.26.0-beta.0`) should be
+backward and forward compatible.
+
+## Feature development
+
+Some features development requires changes in CRI API and corresponding changes
+in _Container Runtime_. Coordinating between Kubernetes branches and release
+versions and _Container Runtime_ versions is not always trivial.
+
+The recommended feature development flow is following:
+
+- Review proposed CRI API changes during the KEP review stage.
+  Some field names and types may not be spelled out exactly at this stage.
+- Locally implement a prototype that implement changes in both - Kubernetes and Container Runtime.
+- Submit a Pull Request for Kubernetes implementing CRI API changes alongside the feature code.
+  Feature must be developed to degrade gracefully when used with older Container Runtime
```

Comment:
Example of what this means in practice on both the kubelet and runtime side will be useful here.